src/Kernel.php

```diff
@@ -123,6 +125,24 @@ private function load(
 
         $this->functionLoader->loadListeners($descriptorsCollection->listenerDescriptors);
 
+        $taskCommand = null;
+        foreach ($descriptorsCollection->taskDescriptors as $taskDescriptor) {
+            $namespace = $taskDescriptor->taskAttribute->namespace;
+            $taskName = $taskDescriptor->taskAttribute->name;
+            $fullTaskName = $namespace ? $namespace . ':' . $taskName : $taskName;
+            if ($fullTaskName === $input->getFirstArgument()) {
+                $taskCommand = $this->taskCommandFactory->createTask($taskDescriptor);
+
+                break;
+            }
+        }
+
+        $taskDefinition = $taskCommand?->getDefinition();
+
+        if (null !== $taskDefinition) {
+            $this->application->getDefinition()->addOptions($taskDefinition->getOptions());
+        }
```

Comment:
I think this part deserve a command in the code.

I think I would also move it to a dedicated private method

Reply:
I fixed a typo :) "command -> comment" :)